### PR TITLE
Toga engine fix and ecam updates 

### LINKED ISSUE #110: ATHR unstable on TOGA
ATHR oscillates between full throttle and down some percent when on TOGA and no other modes like MCT,CMB etc

### PR DESCRIPTION
**Pull request by ThePilot916**
Fixes #110 

**Summary of Changes**(sorry forgot to commit the new file in a separate commit, have attached changes in comments to make things a little simpler)

Fixed ATHR unstable on TOGA
 - Updated engine N1, EGT and Oil pressure values to match A320Neo engine limits.(temperature is in rankine, pressure in pounds per sq feet)
 - N1 and EGT now have dynamic limits based on throttle state(arrow does not turn red beyond 100%N1 on TOGA upto 101.5% for 5min).
 - TOGA state limits on N1 and EGT are only allowed upto 5min on both engines after which it comes down to MCT limits.

APU Guage now features a dynamic EGT warning marker.

Increased gauge px thickness by 1px.

image

### CODE REVIEW

**Comment by Benjozork** [comment]
Please separate changes that do not depend on each other into separate PRs.

**Comment by ThePilot916** [comment]
> Please separate changes that do not depend on each other into separate PRs.

All of them are pretty much dependent.

**Comment by Benjozork** [comment]
> All of them are pretty much dependent.

**Comment by Benjozork** [comment]
> 
> 
> > Please separate changes that do not depend on each other into separate PRs.
> 
> All of them are pretty much dependent.

Alright cheers

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_ECAMGauge.js [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+                marker.setAttribute("y1", start.y.toString());
+                marker.setAttribute("x2", end.x.toString());
+                marker.setAttribute("y2", end.y.toString());
+            }
+        }
+        refreshDangerMinFunction(_value, _force = false){
+            if(_value != this.dangerRange[0]){
+                this.dangerRange[0] = _value;
+            }
+        }
```
Comment (ThePilot916) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_ECAMGauge.js:360]: Added new update function for dynamic markers and dangerMin values.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_ECAMGauge.js [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+                    this.refreshMainValue(this.currentValueFunction());
+                }
+                if (this.outerIndicatorFunction != null) {
+                    this.refreshOuterIndicator(this.outerIndicatorFunction() * 0.01);
+                }
+                if (this.outerDynamicArcFunction != null) {
+                    this.outerDynamicArcFunction(this.outerDynamicArcTargetValues);
+                    this.refreshOuterDynamicArc(this.outerDynamicArcTargetValues[0], this.outerDynamicArcTargetValues[1]);
+                }
+                if(this.outerDynamicMarkerFunction != null) {
```
Comment (ThePilot916) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_ECAMGauge.js:324]: updates for the new dynamic functions.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_ECAMGauge.js [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+                this.refreshOuterIndicator(0, true);
+            }
+            if (this.outerDynamicArcFunction != null) {
+                this.refreshOuterDynamicArc(0, 0);
+            }
+            this.refreshActiveState();
+        }
+
+        //accepts two more parameters to set custom ID for dynamic markers
+        addGraduation(_value, _showInnerMarker, _text = "", _showOuterMarker = false, _setid = false, _idName = "") {
```
Comment (ThePilot916) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_ECAMGauge.js:246]: updated addgraduation to now accept ID to set custom ID for the marker and use the same later with the dynamic function.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_ECAMGauge.js [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+            this.warningRange[1] = _gaugeDefinition.warningRange[1];
+            this.dangerRange[0] = _gaugeDefinition.dangerRange[0];
+            this.dangerRange[1] = _gaugeDefinition.dangerRange[1];
+            this.currentValueFunction = _gaugeDefinition.currentValueFunction;
+            this.currentValuePrecision = _gaugeDefinition.currentValuePrecision;
+            this.outerIndicatorFunction = _gaugeDefinition.outerIndicatorFunction;
+            this.outerDynamicArcFunction = _gaugeDefinition.outerDynamicArcFunction;
+            this.extraMessageFunction = _gaugeDefinition.extraMessageFunction;
+            this.outerDynamicMarkerFunction = _gaugeDefinition.outerDynamicMarkerFunction;
+            this.dangerMinDynamicFunction = _gaugeDefinition.dangerMinDynamicFunction;
```
Comment (ThePilot916) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_ECAMGauge.js:130]: added definitions and init for the new dynamic functions

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_ECAMGauge.css [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    a320-neo-ecam-gauge #RootSVG #GraduationsGroup line.OuterMarker {
+      stroke: #db7200; }
+    a320-neo-ecam-gauge #RootSVG #GraduationsGroup.active {
+      display: block; }
+    a320-neo-ecam-gauge #RootSVG #GraduationsGroup.inactive {
+      display: none; }
+    a320-neo-ecam-gauge #RootSVG #CursorGroup {
+      width: 100%;
+      height: 100%; }
+      a320-neo-ecam-gauge #RootSVG #CursorGroup line {
```
Comment (ThePilot916) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_ECAMGauge.css:60]: increased thickness for line and path on the gauges,

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml [hunk 6]
```diff
@@ -1301,6 +1301,8 @@
 					(A:GEAR HANDLE POSITION, Bool) (A:GEAR POSITION, percent) 99.9 &gt; and (&gt;O:_ShouldBeConnectedToPowerGrid)
 					(O:_ShouldBeConnectedToPowerGrid) sp0
 					2 (&gt;A:BUS LOOKUP INDEX, Number)
+					(A:CIRCUIT CONNECTION ON:21, Bool) l0 != if{ 17 2 (&gt;K:2:ELECTRICAL_BUS_TO_CIRCUIT_CONNECTION_TOGGLE) }
```
Comment (ThePilot916) [inline A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/model/A320_NEO_INTERIOR.xml:1304]: Circuit 17 and 20 are for taxi and take off lights on the nose wheel.

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/engines.cfg [hunk 7]
```diff
@@ -38,13 +38,13 @@ oil_temp_cooling_constant = 0.21
 oil_temp_heating_constant = 700
 oil_temp_tc = 0.03
 oil_temp_tuning_constant = 1
-oil_press_max = 8640.2
+oil_press_max = 18640.2
```
Comment (ThePilot916) [inline A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/engines.cfg:41]: 18640.2 Pounds per sq foot is 130PSI thats supposed to be the gauge max as attached in the image with the PR.

File: A32NX/layout.json [hunk 8]
```diff
@@ -1,619 +1,629 @@
 {
```
Comment (AdenFlorian) [inline A32NX/layout.json]: please dont check this or manifest in anymore, the CI takes care of it, and this jsut creates merge conflicts

**Comment by AdenFlorian** [comment]
It looks like you continued working off a branch that had already been squashed and merged to master. Please open a new PR using a branch that was created of the the latest master, redo your changes on that branch, and have the copied vanilla files added in their own commit. Thanks!

**Pull request closed without merging**